src/components/nfts/NftGrid/index.tsx

```diff
@@ -1,57 +1,204 @@
+import type { SyntheticEvent } from 'react'
+import { useCallback } from 'react'
 import { useMemo, type ReactElement } from 'react'
-import groupBy from 'lodash/groupBy'
-import { Box, Divider, Grid, Typography } from '@mui/material'
+import {
+  Box,
+  Checkbox,
+  InputAdornment,
+  Paper,
+  SvgIcon,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  TextField,
+  Typography,
+} from '@mui/material'
+import FilterAltIcon from '@mui/icons-material/FilterAlt'
+import NftIcon from '@/public/images/common/nft.svg'
 import type { SafeCollectibleResponse } from '@safe-global/safe-gateway-typescript-sdk'
-import NftCard from '../NftCard'
 import ImageFallback from '@/components/common/ImageFallback'
+import ExternalLink from '@/components/common/ExternalLink'
+import useChainId from '@/hooks/useChainId'
+import { nftPlatforms } from '../config'
 
-const NftGrid = ({
-  collectibles,
-  onSendClick,
-}: {
-  collectibles: SafeCollectibleResponse[]
-  onSendClick?: (nft: SafeCollectibleResponse) => void
-}): ReactElement => {
-  const collections: Record<string, SafeCollectibleResponse[]> = useMemo(
-    () => groupBy(collectibles, 'address'),
-    [collectibles],
+interface NftsTableProps {
+  nfts: SafeCollectibleResponse[]
+  selectedNfts: SafeCollectibleResponse[]
+  onSelect: (item: SafeCollectibleResponse) => void
+  onFilter: (value: string) => void
+}
+
+const minRows = 10
+const iconSize = 20
+const iconStyle = { width: '100%', maxHeight: '100%' }
```

Comment:
This stretches some images if their original height is bigger than their width.

For instance:
This image
![Original](https://assets.otherside.xyz/otherdeeds/790b73738484af857ab26788630164ed8ed09ce5e582c79a4cc177000a5b5481.jpg)

Looks like this
![Screenshot 2023-01-19 at 16 22 59](https://user-images.githubusercontent.com/2670790/213481892-b6234ea3-21ad-4af6-9fa4-566ba2ef69ef.png)